src/hooks/useGroupedData.ts

```diff
@@ -9,19 +9,34 @@ function useGroupedData<T extends Record<string, unknown>>(
       return [];
     }
 
-    const groupedData = data?.reduce((acc: Record<string, T[]>, item: T) => {
+    const groupedData = data.reduce((acc: Record<string, T[]>, item: T) => {
       const groupKey = item[key];
       if (!acc[groupKey as string]) {
```

Comment:
No type assertions please.

Reply:
Sir @Monfernape , No need for this hook anymore because I've fixed it local file.